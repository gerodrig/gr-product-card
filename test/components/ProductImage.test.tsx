import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components'
import { product2 } from '../data/products';


describe('tests in ProductImage component', () => {
    
    test('should render ProductTitle component with the custom title', () => {
        const wrapper = renderer.create(<ProductImage img='./test.jpg' />);

        expect(wrapper.toJSON()).toMatchSnapshot();
    }
    );

    test('should show the image path', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage img={product2.img} style={{ backgroundColor: 'black'}} />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
    
});

