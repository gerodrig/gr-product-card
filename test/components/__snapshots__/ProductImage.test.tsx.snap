// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in ProductImage component should render ProductTitle component with the custom title 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="./test.jpg"
/>
`;

exports[`tests in ProductImage component should show the image path 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  />
</div>
`;
